src/hooks/useTodoController.ts

```diff
@@ -0,0 +1,276 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { Todo } from '../types/todo/Todo';
+import * as todoService from '../api/todos';
+import { ErrorMessage } from '../enums/errorMessage';
+import { StatusFilter } from '../enums/statusFilter';
+import { getFilteredTodos } from '../utils/getFilteredTodos';
+
+export const useTodoController = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isTodosLoading, setIsTodosLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState<ErrorMessage>(
+    ErrorMessage.None,
+  );
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAddTodoFormFocused, setIsAddTodoFormFocused] = useState(false);
+  const [todoToDeleteIds, setTodoToDeleteIds] = useState<Todo['id'][]>([]);
+  const [todosToUpdate, setTodosToUpdate] = useState<Todo[]>([]);
```

Comment:
```suggestion
  const [todosToUpdate, setTodosToUpdate] = useState<Todo[]>([]);
```
same here do we need this state?
can we use todo array to find editing todo?
change state -- it's rerenders, you killed FE performance